app/models/vacancy/publish.rb

```diff
@@ -0,0 +1,10 @@
+class Vacancy::Publish
+  def initialize(vacancy)
+    @vacancy = vacancy
+  end
+
+  def publish
+    @vacancy.save!
+    NotifyVacancyOnSlackWorker.perform_async(@vacancy.id)
```

Comment:
You can pass the Worker as a dependency, it will make your tests easier, because instead of count the jobs, you can check if the method was called, something like that: 

``` 
class Vacancy::Publish
  def initialize(worker = NotifyVacancyOnSlackWorker)
   ...
  end

#in your tests  you can make like: 

Vacancy::Publish.new(spy).publish(vacancy)
#here you check if your spy.perform_async  method was called.